webapp/publisher/snaps/views.py

```diff
@@ -1390,11 +1404,16 @@ def get_snap_builds(snap_name):
                 }
             )
     else:
-        github = GitHubAPI(flask.session.get("github_auth_secret"))
-        context["github_user"] = github.get_user()
-
-        if context["github_user"]:
-            context["github_repositories"] = github.get_user_repositories()
+        try:
+            context["github_user"] = github.get_username()
+            # Get snapcraft repositories sorted by snapcraft_yaml
+            context["github_repositories"] = sorted(
+                github.get_user_repositories(),
+                key=lambda k: k["snapcraft_yaml"],
+                reverse=True,
+            )
+        except Unauthorized:
+            pass
```

Comment:
Why wouldn't we care about an unauthorized response?

Reply:
If we get an Unauthorized exception here that means the user is not logged in GitHub, that means github_user will be None in the template and we will show the login button.

Reply:
Then maybe you could just put the line in here `context["github_user"] = None`? Just to show that we are deliberately doing something.

Reply:
Yes, it will make it easier to understand